scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+		{
+			//qDebug() << "Newline: ";
+			if (pass == FrameworkLineTests::FIRSTPOINT)
+			{
+				if (textRegionLines.empty()) {
+					textRegionLines.push_back(TextRegionLine());
+				}
+			} 
+			else
+			{
+				textRegionLines.push_back(TextRegionLine());
+			}
+			textRegionLine = &textRegionLines.back();
+			textRegionLine->baseOrigin = newPoint;
+			if (pass == FrameworkLineTests::NEWLINE)
+			{
+				textRegionLine->maxHeight = abs(newPoint.y() - lastXY.y());
+				if (textRegionLines.size() == 2)
+				{
+					lineSpacing = abs(newPoint.y() - lastXY.y()) + 1;
+					//qDebug() << "setting lineSpacing to:" << lineSpacing;
+				}
+			}
+		}
+
+		textRegionLine = &textRegionLines.back();
+		if ((pass == FrameworkLineTests::FIRSTPOINT && textRegionLine->segments.empty()) || pass == FrameworkLineTests::NEWLINE ||  pass != FrameworkLineTests::FIRSTPOINT && textRegionLine->segments[0].glyphIndex != textRegionLine->glyphIndex)
+		{
+			TextRegionLine newSegment = TextRegionLine();
+			textRegionLine->segments.push_back(newSegment);
+		}
+		TextRegionLine* segment = &textRegionLine->segments.back();
+		segment->baseOrigin = newPoint;
+		if (pass == FrameworkLineTests::STYLESUPERSCRIPT)
+		{
+			segment->maxHeight = abs(lineSpacing - (newPoint.y() - lastXY.y()));
+		
+		}
+		else
+		{
+			segment->maxHeight = textRegionLines.back().maxHeight;
+		}
+
+		if (pass != FrameworkLineTests::NEWLINE && pass != FrameworkLineTests::FIRSTPOINT)
+		{
+			textRegionLines.back().segments.back().width = abs(textRegionLines.back().segments.back().baseOrigin.x() - newPoint.x());					
+			textRegionLine = &textRegionLines.back();
+			textRegionLine->width = abs(textRegionLine->baseOrigin.x() - newPoint.x());
+		}
+		
+		maxHeight = abs(textRegioBasenOrigin.y() - newPoint.y()) > maxHeight ? abs(textRegioBasenOrigin.y() - newPoint.y()) : maxHeight;
+		lastXY = newPoint;
+	}
+	else
+	{
+		//qDebug() << "movetopoint failed";
+	}
+
+	return pass;
+}
+
+//TODO:, extract some font heights instesad of using dx all the time
+TextRegion::FrameworkLineTests TextRegion::addGlyphAtPoint(QPointF newGlyphPoint, PdfGlyph newGlyph)
+{
+	QPointF movedGlyphPoint = QPointF(newGlyphPoint.x() + newGlyph.dx, newGlyphPoint.y() + newGlyph.dy);
+	//qDebug() << "addGlyphAtPoint start" << newGlyphPoint << " glyph:"<< new_glyph.code;
+	//FIXME: There should be no need for testing the scope when adding a new glyph because moveto should have been called first but leave it in for now to catch any errors in the logic
+	if (glyphs.size() == 1)
+	{
+		// FIXME: do a propper lookup of the height
+		lineSpacing = newGlyph.dx * 2;
+		//qDebug() << "addGlyphAtPoint start";
+		lastXY = newGlyphPoint;
+		lineBaseXY = newGlyphPoint;
+	}	
+	bool xInLimits = false;
+	if (closeToX(newGlyphPoint.x(), lastXY.x()))
+	{
+		xInLimits = true;
+	}
+	bool yInLimits = false;
+	if (closeToY(newGlyphPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newGlyphPoint, xInLimits, yInLimits);

```

Comment:
again, not more than three lines needed here...

Reply:
since it's repeated multiple time, the check to closeness and the linear test should be in a separate function that better says what the value of pass is! (is it one value, multiple ones?)